terraform/firewall/variables.tf

```diff
@@ -0,0 +1,130 @@
+variable "basename" {
+  type        = string
+  description = "Basename of the module."
+  validation {

```

Comment:
Replace with the following validation:

validation {
condition = can(regex("^[-\.\w]{1,52}$", var.basename)) && can(regex("[\w]+$", var.basename))
error_message = "The name must be between 1 and 52 characters, must end with a letter, number or underscore, and may contain only letters, numbers, underscores, periods, or hyphens."
}